pkg/compute/kube/client_test.go

```diff
@@ -2500,6 +2514,71 @@ func TestK8sServicesToMeshServices(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "duplicate ports on k8s service with different protocols",
+			svc: corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns1",
+					Name:      "s1",
+				},
+				Spec: corev1.ServiceSpec{
+					ClusterIP: "10.0.0.1",
+					Ports: []corev1.ServicePort{
+						{
+							Name:     "p1",
+							Port:     80,
+							Protocol: corev1.ProtocolTCP,
+						},
+						{
+							Name:     "p2",
+							Port:     80,
+							Protocol: corev1.ProtocolUDP,
+						},
+					},
+				},
+			},
+			svcEndpoints: []runtime.Object{
+				&corev1.Endpoints{
+					ObjectMeta: metav1.ObjectMeta{
+						// Should match svc.Name and svc.Namespace
+						Namespace: "ns1",
+						Name:      "s1",
+					},
+					Subsets: []corev1.EndpointSubset{
+						{
+							Ports: []corev1.EndpointPort{
+								{
+									// Must match the port of 'svc.Spec.Ports[0]'
+									Name: "p1",
+									Port: 8080, // TargetPort
+								},
+								{
+									// Must match the port of 'svc.Spec.Ports[1]'
+									Name: "p2",
+									Port: 8080, // TargetPort
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: []service.MeshService{
+				{
+					Namespace:  "ns1",
+					Name:       "s1",
+					Port:       80,
+					TargetPort: 8080,
+					Protocol:   "http",
+				},
+				{
+					Namespace:  "ns1",
+					Name:       "s1",
+					Port:       80,
+					TargetPort: 8080,
+					Protocol:   "",
```

Comment:
I don't know if we want a blank protocol to be the end-result here; don't we want to preserve the original protocol and make Envoy error later down the stack?

Reply:
We should leave this blank as the app protocol is unknown. The app protocol is not to be confused with the protocol field. We want to preserve the app protocol, not the protocol. If it's unknown, we should leave it as is for non TCP case.

Reply:
Ah, I see what you're saying; we're just not going to make OSM aware of anything non-TCP. Got it 👍🏾